Cargo.toml

```diff
@@ -25,7 +26,7 @@ rust-version = "1.70"
 [workspace.dependencies]
 diff = { version = "0.1.12", default_features = false }
 regex = { version = "1.3", default_features = false, features = ["std"] }
-regex-syntax = { version = "0.8", default_features = false }
+regex-syntax = { version = "0.8.2", default_features = false }
```

Comment:
Do we actually need to specify this here?  If the `.2` support is just a dependency of logos, I think specifying only `0.8` on our end will result in feature unification demanding the right version when it build logos.  The advantage to only specifying `0.8` being that if something about logos were to change, we get that automatically, and could in theory support (for example) 0.8.1 environments if logos were to drop their requirement.

On my box, only specifying 0.8 works fine and pulls in the 0.8.2 version as expected.

Reply:
It got caught up in the min_versions workflow check. Unfortunately, we can't have dev-dependency members or something like that in the workspace(since we really only want the 0.8.2 version for testing). I think we could: disable that check, remove the lexer example form the workspace members and run it manually, ask the logos people to use the minimum version on their end or just stick with 0.8.2 since it's a patch release which should be trivial upgradable for someone in a 0.8.1 environment(and if it isn't, that's a regex bug).

Reply:
Yuck.  I'm inclined to call this a bug in -Zdirect-minimal-versions, personally.  (I found someone trying to report it here: https://github.com/rust-lang/cargo/issues/5657#issuecomment-1914446399, and it didn't seem like epage agreed that it was a bug, although I don't think the reporters desired behavior was what I was expecting either).

Regardless, I imagine it's messy/undoable to get minimal versions and dependency resolution playing nicely (I'd expect it to ultimately resolve to 0.8.2 here, since that what's actually minimal required *after* feature resolution), so this is probably not a battle worth fighting. I'd rather just set our version to 0.8.2 than any of the workaround you mention.